aws_connect.py

```diff
@@ -1,45 +1,57 @@
-import sys
 import boto3
 import configparser
 import botocore.exceptions
+from fastapi import FastAPI, HTTPException
 
 
-def aws_connect(config_file_path: str, action: str):
-    config = configparser.ConfigParser()
-    config.read(config_file_path)
-    try:
-        session = boto3.Session(
-            aws_access_key_id=config['CREDENTIALS']['aws_access_key_id'],
-            aws_secret_access_key=config['CREDENTIALS']['aws_secret_access_key'],
-            region_name=config['CREDENTIALS']['region_name']
-        )
-        ec2 = session.resource('ec2')
-        instance = ec2.Instance(config['CREDENTIALS']['instance_id'])
-    except KeyError:
-        print(f"{config_file_path} does not exist, is of incorrect type, has incorrect structure or contains "
-              f"incorrect credential names")
-        return
+app = FastAPI()
+config = configparser.ConfigParser()
+config.read("config.ini")
+try:
+    session = boto3.Session(
+        aws_access_key_id=config['CREDENTIALS']['aws_access_key_id'],
+        aws_secret_access_key=config['CREDENTIALS']['aws_secret_access_key'],
+        region_name=config['CREDENTIALS']['region_name']
+    )
+    ec2 = session.resource('ec2')
+except KeyError:
+    print(f"config.ini does not exist, is of incorrect type, has incorrect structure or contains "
+          f"incorrect credential names")
+
+
+@app.get("/instances/{instance_id}/status")
+async def status(instance_id: str):
     try:
-        if action == "start":
-            instance.start()
-        elif action == "stop":
-            instance.stop()
-        elif action == "state":
-            print(instance.state)
-        else:
-            print("Available actions: start, stop, state")
+        instance = ec2.Instance(instance_id)
+        return instance.state
     except botocore.exceptions.ClientError as err:
-        print(err)
-        return
+        if err.response["Error"]["Code"] == "InvalidInstanceID.Malformed":
+            raise HTTPException(status_code=404, detail="Incorrect instance id")
+        else:
+            return err
 
 
-def main():
-    if len(sys.argv) != 3:
-        print("Usage: ./aws_connect <config_file_path> <action>")
-    else:
-        aws_connect(sys.argv[1], sys.argv[2])
-    return 0
+@app.get("/instances/{instance_id}/start")
+async def start(instance_id: str):
+    try:
+        instance = ec2.Instance(instance_id)
+        instance.start()
+        raise HTTPException(status_code=204)
```

Comment:
exceptions should be used for error paths, the successful 204 code can be achieved by other means: https://fastapi.tiangolo.com/tutorial/response-status-code/